### ISSUE TITLE
pkg.mxe.cc down?

### ISSUE THREAD

**Issue by fejoa** [mxe/mxe#2253]
Hello from the IVAN community at https://github.com/Attnam/ivan

We have been building our IVAN windows binaries using mxe for about a year now. We are proud to use it, and we truly appreciate all the great people who make mxe possible. 

We have recently come across this: https://travis-ci.org/Attnam/ivan/jobs/467359638
Is the mxe package server down for maintenance? If so, when will it be back up?

**Comment by tonytheodore** [comment]
It does seem to be down, @starius can you have a look?

**Comment by belovachap** [comment]
Our build is also broken :( https://travis-ci.org/peercoin/peercoin/builds/468123051

**Comment by tonytheodore** [comment]
Unfortunately, there aren't any alternative servers so I've started a build on an Ubuntu Trusty machine to at least get the packages built (takes a few days, all else being equal).

I haven't set up an apt repo before, but I think they can be hosted on s3. There's an open issue to get more regular/recent pkg builds (#2021), moving to s3 (or similar) should help with both issues.

**Comment by fejoa** [comment]
Thanks again @tonytheodore, that's good news. Good luck with the build, it sounds like a mammoth task by all accounts; please do let us know how it goes : )

**Comment by tonytheodore** [comment]
The qt5 build is broken on trusty with gcc 4.8 but there's a fix https://bugreports.qt.io/browse/QTBUG-71945.

I have some debs now to get started with setting up a repo.

**Comment by tehnick** [comment]
@tonytheodore add label `build-pkg` to this issue please.

**Comment by tonytheodore** [comment]
Getting closer, I can create a bare minimum repo that can be fetched with Ubuntu 16.04, but not with 14.04 (trusty). I can't actually get trusty to fetch from a repo it creates itself so far...

**Comment by tehnick** [comment]
Personally I think that support of Ubuntu Trusty is just wasting of time now. As for Travis CI, this is just a question of one string in `yml` config (`dist: xenial`) and all will work fine.

**Comment by tonytheodore** [comment]
[This tutorial](https://www.digitalocean.com/community/tutorials/how-to-use-reprepro-for-a-secure-package-repository-on-ubuntu-14-04) works using completely new installs of Trusty so it must be something about my current VMs.

**Comment by tonytheodore** [comment]
Grrr, turns out to be gnutls (via apt) on Trusty not liking the https setup I was using. It's really not worth time to make that work.

**Comment by tonytheodore** [comment]
So now I can build debs on other machines (with tweaks from #2256) and add them to the repo:
```bash
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 86B72ED9
sudo add-apt-repository 'deb [arch=amd64] http://mirror.mxe.cc/repos/apt trusty main'
# or https for xenial
# sudo add-apt-repository 'deb [arch=amd64] https://mirror.mxe.cc/repos/apt xenial main'
sudo apt-get update
apt-cache search mxe-
```
They're only minimal builds so far, please test and let me know how it goes.

**Comment by tehnick** [comment]
> please test and let me know how it goes.

I have tested in Ubuntu Xenial on Travis CI:

> sudo add-apt-repository 'deb [arch=amd64] https://mirror.mxe.cc/repos/apt xenial main'

Some packages are [absent](https://travis-ci.org/eiskaltdcpp/eiskaltdcpp/jobs/472127401) now:
```
+sudo apt-get install -qq mxe-i686-w64-mingw32.shared-aspell mxe-i686-w64-mingw32.shared-boost mxe-i686-w64-mingw32.shared-jsoncpp mxe-i686-w64-mingw32.shared-libidn mxe-i686-w64-mingw32.shared-lua mxe-i686-w64-mingw32.shared-miniupnpc mxe-i686-w64-mingw32.shared-qtmultimedia mxe-i686-w64-mingw32.shared-qttools
E: Unable to locate package mxe-i686-w64-mingw32.shared-aspell
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-aspell'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-aspell'
E: Unable to locate package mxe-i686-w64-mingw32.shared-boost
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-boost'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-boost'
E: Unable to locate package mxe-i686-w64-mingw32.shared-jsoncpp
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-jsoncpp'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-jsoncpp'
E: Unable to locate package mxe-i686-w64-mingw32.shared-libidn
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-libidn'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-libidn'
E: Unable to locate package mxe-i686-w64-mingw32.shared-lua
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-lua'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-lua'
E: Unable to locate package mxe-i686-w64-mingw32.shared-miniupnpc
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-miniupnpc'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-miniupnpc'
E: Unable to locate package mxe-i686-w64-mingw32.shared-qtmultimedia
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-qtmultimedia'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-qtmultimedia'
E: Unable to locate package mxe-i686-w64-mingw32.shared-qttools
E: Couldn't find any package by glob 'mxe-i686-w64-mingw32.shared-qttools'
E: Couldn't find any package by regex 'mxe-i686-w64-mingw32.shared-qttools'
The command "./tests/travis/install-build-depends.sh" failed and exited with 100 during .
```

> add-apt-repository 'deb [arch=amd64] http://mirror.mxe.cc/repos/apt trusty main'

It [works](https://travis-ci.org/eiskaltdcpp/eiskaltdcpp/jobs/472130479) like a charm.

Thank you!

**Comment by tonytheodore** [comment]
Xenial and Stretch were just minimal builds to test. They should be finished full builds in a few hours.

**Comment by tonytheodore** [comment]
Xenial, Stretch, and Trusty now have full repos (sans itk-i686-w64-mingw32.shared for all and ghostscript-x86_64-w64-mingw32.shared for Trusty).

Note this isn't final - I still have to sort out:
  - repo layout
  - permanent signing keys
  - location and DNS records of new server

**Comment by fejoa** [comment]
@tonytheodore thanks to your new mirror, we have our [build functioning again](https://travis-ci.org/Attnam/ivan/jobs/483645966) via [same changes](https://github.com/Attnam/ivan/commit/7caa853799f7ee11612c0febc0f732d8c7106105) as @tehnick [above](https://github.com/mxe/mxe/issues/2253#issuecomment-449868798). Thank you to you both.

We will revert the change and stand down our mxe builds for now, as the test confirms we have everything we need. Can you please let us know once the mirror is ready for general use?

**Comment by Lorac** [comment]
@tonytheodore any chance to support debian jessie?

**Comment by singpolyma-shopify** [comment]
is pkg.mxe.cc coming back?  If not, should the docs on the website be updated to point at this mirror?

**Comment by tonytheodore** [comment]
@Lorac I plan to support Ubuntu LTS (trusty, xenial, bionic) and Debian old stable, stable, and testing (jessie, stretch, buster). VTK6 was removed 18 months (#1852) and we don't have the time to support multiple versions. 

I do plan to write some notes on how to use mxe pre-built pkgs to bootstrap your own apt repo, but I have to work out how to do that first ;)

@vog could you point pkg.mxe.cc to the same address as mirror.mxe.cc? The old site is on the [wayback machine](https://web.archive.org/web/20180906074049/http://pkg.mxe.cc/) and I'll copy that over and update the docs.

**Comment by mbunkus** [comment]
Trusty's support is supposed to end next month. Do you really think it worth supporting it now?

**Comment by vog** [comment]
@tonytheodore I changed the DNS configuration as requested.

Old DNS settings:

```
mirror                   IN A       104.131.71.203
pkg                      IN A       51.15.11.129
mirror                   IN AAAA    2604:a880:800:10::156:f001
pkg                      IN AAAA    2001:bc8:338c::1
```

New DNS settings:

```
mirror                   IN A       104.131.71.203
pkg                      IN A       104.131.71.203
mirror                   IN AAAA    2604:a880:800:10::156:f001
pkg                      IN AAAA    2604:a880:800:10::156:f001
```

It might theoretically take up to 24 hours for the new settings to become active, although I'm confident that within 2 hours everything will be up to date.

**Comment by tonytheodore** [comment]
Thanks @vog I can see it already.

@mbunkus trusty is still the [default on Travis CI](https://docs.travis-ci.com/user/reference/overview/#linux) and these repos take less than 5GB space so it's not an issue to support it.

**Comment by tehnick** [comment]
@tonytheodore 
> I plan to support Ubuntu LTS (trusty, xenial, bionic) and Debian old stable, stable, and testing (jessie, stretch, buster).

I would suggest you to not waste time and disk space for Debian Jessie. It may become old-old stable quite soon. (Probably in this year.)

**Comment by ams-tschoening** [comment]
I was running into this issue as well today and fixed it by removing all `mxe-*`-packages, adding the new mirror as repo for APT and reinstalling the packages I need for the project `fwbuilder`. The problem is that now some files are missing the project depends on and I'm pretty sure those have been available in the past, else building would have failed in the past as well.

    Processing script file: "release/install-root/fwbuilder.nsi"
    File: "/usr/lib/mxe/usr/i686-w64-mingw32.shared/bin/ssleay32.dll" -> no files found.

Is the missing file related to this problem and not provided yet/anymore or have packages changed and that file comes from somewhere else, not installed anymore automatically? Is it something completely different and I should file some independent issue?

**Comment by tehnick** [comment]
@ams-tschoening 
IIRC OpenSSL was updated in MXE. And new version of OpenSSL consist of less amount of libraries.

**Comment by tehnick** [comment]
> IIRC OpenSSL was updated in MXE. And new version of OpenSSL consist of less amount of libraries.

I have re-checked: `ssleay32.dll` was renamed to `libssl-1_1.dll` after updating from `1.1.0*` to `1.1.1*`. And 64-bit version of shared library is now called `libssl-1_1-x64.dll`.

**Comment by ams-tschoening** [comment]
Thanks, I recognized `libssl-1_1.dll` already, but wasn't sure if this is something completely different. There are additional renames and I provided a PR to [fwbuilder](https://github.com/fwbuilder/fwbuilder/pull/83).

**Comment by ooxi** [comment]
Please update the instructions on how to use the prebuild binaries. They are very useful, but we cannot expect new users to comb through GitHub issues in order to find out how to use them.

**Comment by vog** [comment]
@ooxi Would you mind proposing a concrete text change and/or pull request? (Documentation improvements are just pull requests like any other change, mostly to the file `docs/index.html`)

**Comment by vog** [comment]
Why was this closed? A reference to e.g. another ticket explaining this would be fine.

**Comment by tonytheodore** [comment]
I think I started writing #2535 and got distracted!

It's all working now but would be hard to replicate.
